### DISCUSSION TITLE
Is there any reason why the Role object does not have fields corresponding to its Permissions ?

### DISCUSSION THREAD

**Discussion by mlbiche** [directus/directus#16670] [Feature Requests]
Is there any reason why the Role object does not have fields corresponding to its Permissions ?

It would simplify requests on the role page in the admin application by loading everything in one request.

It would also simplify requesting a user permission by using `role.permissions.*` value for the `field` query param.

**Comment by br41nslug** [comment]
Can you elaborate on how you envision this? Im having trouble seeing what approach exactly you're suggesting because fields correlate to database columns and permissions are in a related table to roles but not directly part of it.

**Reply by mlbiche** [reply]
If permissions were part of roles (like a relational field is when the "Corresponding field" checkbox is checked when creating it), they could be loaded along roles.

For example, when accessing `/admin/settings/roles/my-role-id`, instead of 2 requests to `/roles/my-role-id` and to `/permissions?filter={"role": { "_eq": "my-role-id" } }&limit=-1`, we would only have one request to `/roles/my-role-id?fields=*,permissions.*`.

In a WebApp using Directus as an API, it would be also easier to turn on/off some features based on permissions. Instead of writing conditions based on the role id/name of the authenticated user, or querying Directus to get the expected permission, it could be load directly after authentication when loading the authenticated user through `/users/me?fields=*,role.permissions.*`.

Don't know if it is clearer ?

**Reply by br41nslug** [reply]
The relation (all be it one way from permissions to roles) already exists so adding the reverse definition is very possible. That would make the permissions queryable in one call from roles (or users for that matter). 
The problem i see with this approach however is performance because the permissions data grows with the created collections (and some userd have enormous databases) fetching this separately was presumably a performance consideration to keep the app most responsive.

**Reply by mlbiche** [reply]
Ok, seems legit to me 👍 
Then, could this be updated in the collection schema, without updating the `/admin/settings/roles/my-role-id` page in the Admin app, so WebApp using Directus as an API could benefit of this shortcut if they would ? Cheers

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
